src/s3/fileViewerManager.ts

```diff
@@ -88,12 +90,13 @@ export class S3FileProvider implements FileProvider {
         return telemetry.s3_uploadObject.run(async span => {
             span.record({ component: 'viewer' })
 
+            const mimeType = mime.contentType(path.extname(this._file.name)) || undefined
             const result = await this.client
                 .uploadFile({
                     content,
                     key: this._file.key,
                     bucketName: this._file.bucket.name,
-                    contentType: mime.contentType(path.extname(this._file.name)) || undefined,
+                    contentType: this._file.ContentType ?? mimeType,
```

Comment:
out of scope for this PR, but as a future todo: can/should `downloadFile` get the current mimetype stored in S3 and re-use that in write()?